app/scripts/controllers/app-state-controller.ts

```diff
@@ -179,89 +184,239 @@ const getDefaultAppStateControllerState = (
   newPrivacyPolicyToastClickedOrClosed: null,
   newPrivacyPolicyToastShownDate: null,
   hadAdvancedGasFeesSetPriorToMigration92_3: false,
-  ...initState,
-  qrHardware: {},
-  nftsDropdownState: {},
   surveyLinkLastClickedOrClosed: null,
-  signatureSecurityAlertResponses: {},
-  switchedNetworkDetails: null,
   switchedNetworkNeverShowMessage: false,
-  currentExtensionPopupId: 0,
+  ...getDefaultAppStateOverrides(),
 });
 
-export class AppStateController extends EventEmitter {
-  private readonly extension: AppStateControllerOptions['extension'];
+function getDefaultAppStateOverrides() {
```

Comment:
This seems like the default state to me, what are your thoughts on renaming this `getDefaultAppStateControllerState` for consistency with other controllers?

``` suggestion
function getDefaultAppStateControllerState() {
```

Reply:
We already have getDefaultAppStateControllerState. Check here for context https://github.com/MetaMask/metamask-extension/pull/28784/files#diff-037120a57cb346b20107f8eea73f9fc79a23738050db9828196e0eefdd8ffc00R380

getDefaultAppStateOverrides is created to address https://github.com/MetaMask/metamask-extension/pull/28784#discussion_r1862808829

Reply:
Gotcha, I see.

Perhaps we ought to name this `getInitialStateOverrides`? Since it's the initial state that gets overridden, not the default state:

``` suggestion
function getInitialStateOverrides() {
```